bot/cogs/allow_deny_lists.py

```diff
@@ -31,6 +31,24 @@ async def _add_data(
         """Add an item to an allow or denylist."""
         allow_type = "whitelist" if allowed else "blacklist"
 
+        # If this is a server invite, we gotta validate it.
+        if list_type == "GUILD_INVITE":
+            log.trace(f"{content} is a guild invite, attempting to validate.")
+            validator = ValidDiscordServerInvite()
```

Comment:
If I understand this correctly, you only allow invite links, but you end up storing the guild ID in the DB? Why didn't you accept guild IDs directly? Is it because you wouldn't be able to auto-generate a comment?

Reply:
Yes. Guild IDs cannot be looked up in order to get any kind of guild information (for data privacy reasons). We can't just store the invites either, because we want to whitelist an entire guild, not just a specific invite.

Reply:
Sounds reasonable, thanks.